R-package/R/lgb.Dataset.R

```diff
@@ -93,20 +114,45 @@ Dataset <- R6::R6Class(
 
     create_valid = function(data,
                             info = list(),
+                            label = NULL,
+                            weight = NULL,
+                            group = NULL,
+                            init_score = NULL,
+                            params = list(),
                             ...) {
 
+      additional_params <- list(...)
+      if (length(additional_params) > 0L) {
+        warning(paste0(
+          "Dataset$create_valid(): Found the following passed through '...': "
+          , paste(names(additional_params), collapse = ", ")
+          , ". These will be used, but in future releases of lightgbm, this warning will become an error. "
+          , "Add these to 'params' instead. "
+          , "See ?lgb.Dataset.create.valid for documentation on how to call this function."
+        ))
+      }
+
+      # anything passed into '...' should be overwritten by things passed to 'params'
+      params <- modifyList(additional_params, params)
+
+      # the Dataset's existing parameters should be overwritten by any passed in to this call
+      params <- modifyList(private$params, params)
```

Comment:
oh I get `get_params()` would probably be better. Neither should be wrong, and I just used `private$params` because that's what was already being used, but I agree `get_params()` should be slightly preferred.

Reply:
```suggestion
      params <- modifyList(private$get_params(), params)
```